tools/sideloading/sideloading_agent.py

```diff
@@ -0,0 +1,280 @@
+import sys
+import os
+import time
+import datetime
+import urllib
+import json
+import threading
+import http.server
+import ssl
+import subprocess
+
+
+# ---
+
+sideloaded_dir = "/opt/aws/panorama/storage/sideloaded"
+panorama_default_dir = "/panorama"
+
+# cert / key files have to be located at the same directory as this script.
+cert_key_dir = os.path.dirname(__file__)
+
+# ---
+
+class ApplicationProcessManager:

```

Comment:
add extensive logging for API request/responses in sideloading_agent

Reply:
Adding log_message method in the SideloadingRequestHandler to customize log format and flush to stdout.

Reply:
Output confirmed:

```
2022-11-11T22:36:42.155670871+00:00 stdout F SideloadingAgent started
2022-11-11T22:52:11.185369443+00:00 stdout F SideloadingRequest : "GET /files HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.689874890+00:00 stdout F SideloadingRequest : "GET /files HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.728066760+00:00 stdout F SideloadingRequest : "PUT /files/.keep HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.757662967+00:00 stdout F SideloadingRequest : "PUT /files/app.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.782064406+00:00 stdout F SideloadingRequest : "PUT /files/boot.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.811482660+00:00 stdout F SideloadingRequest : "PUT /files/cv_app.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.839148673+00:00 stdout F SideloadingRequest : "PUT /files/cv_app_1.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.869819226+00:00 stdout F SideloadingRequest : "PUT /files/cv_app_2.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.895435588+00:00 stdout F SideloadingRequest : "PUT /files/cw_post_metric.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.920169927+00:00 stdout F SideloadingRequest : "PUT /files/log_utils.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.947855620+00:00 stdout F SideloadingRequest : "PUT /files/message_utils.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:40.975505922+00:00 stdout F SideloadingRequest : "PUT /files/metrics.py HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:41.181173116+00:00 stdout F SideloadingRequest : "PUT /files/movenet_multipose_lightning_1/saved_model.pb HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:41.573092737+00:00 stdout F SideloadingRequest : "PUT /files/movenet_multipose_lightning_1/variables/variables.data-00000-of-00001 HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:41.598819916+00:00 stdout F SideloadingRequest : "PUT /files/movenet_multipose_lightning_1/variables/variables.index HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:52:49.647976521+00:00 stdout F SideloadingRequest : "GET /files HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:53:01.386588441+00:00 stdout F SideloadingRequest : "POST /application HTTP/1.1" 200 - : 10.0.0.154
2022-11-11T22:53:01.469883604+00:00 stdout F Hello from sideloaded cv_app.py 4
```